raft-tech/TANF-app
# 2038 - CSRF settings Bug fix for non-production sites

## Summary of Changes
Pull request closes #2038 

Ensuring that the proper environment variables, middleware, and session settings were set for the environment that was being deployed to.

## How to Test

1. Make sure both staging and production are deployed from this branch, which may require modifying the CI.
2. Test with Google Chrome as it seems to fail more often then Firefox.
3. Use an incognito session or clear all session history.
4. Attempt to login to both staging and production
5. Double check the links when logging in to make sure they are all correct for the environment.

![Screen Shot 2022-08-22 at 3 03 02 PM](https://user-images.githubusercontent.com/7119690/186026809-98e60050-9b08-4fd4-85ad-13a066b3291c.png)

This video shows the clam av file scans still working.
https://user-images.githubusercontent.com/7119690/186338144-c8440356-af5d-4121-9288-ecdd9a46a28d.mp4



## Deliverables

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] Production environments utilizing `*.acf.hhs.gov` domains retain frontend button functionality (login, request access, admin panel)
- [x] Non-production environments utilizing `*.app.cloud.gov` domains retain their previous functionality (login, request access, admin panel)
- [ ] CORS best practices are implemented

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [ 96.71% ](https://app.codecov.io/gh/raft-tech/TANF-app/branch/main)
  + [x] Backend coverage: [93.84%](https://app.codecov.io/gh/raft-tech/TANF-app/branch/main)

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
